### ISSUE TITLE
Feat: Pool Fees - Charge on Cash asset directly

### ISSUE THREAD

**Issue by mustermeiszer** [centrifuge/centrifuge-chain#1757]
### Research/based on
→ Create fee assets with target `Reserve`/`CashAsset`
→ Charging on `CashAsset` emits `Paid` event
→ Uncharging on `CashAsset` emits `UnPaid `event?
   * Subquery needs to reduce total paid with receiving Unpaid events
MAYBE:
Charge at most balance of CashAsset
Have a limit in DOLLAR - not compared to nav
Needs to have a limit per time parsed based on calender time
Date based - E.g.: if fee is created in december, you are still able to charge the fee for the whole year in the december

### How will this affect the code base

[Does it improve readability, performance? Will it introduce new complexity?]

### What are foreseen obstacles or hurdles to overcome?

[Are migrations needed, structural changes around testing etc?]

**Comment by wischli** [comment]
IIUC, these are more or less "one time" fees in that they can be removed once the USD limit is reached. If so, it doesn't make sense to couple them with a time component (e.g. yearly).

Unforeseen issues: 
* Requires oracle for CFG to USD conversion

**Comment by mustermeiszer** [comment]
> IIUC, these are more or less "one time" fees in that they can be removed once the USD limit is reached. If so, it doesn't make sense to couple them with a time component (e.g. yearly).

Why? The limit is per period, but compounding over periods. So daily 50EUR would mean you can charge 250EUR after 5 days if you haven't charged anything in between.

> Requires oracle for CFG to USD conversion

Why do we need that? Charges are in pool currency?

**Comment by mustermeiszer** [comment]
> My question would be: Should these fees support completely flexible time bounds or is annually and potentially monthly sufficient?

Yes, flexible timebounds would be greatl Just to that we can make the interval based on the calendar - I think we get all if we allow to choose
* x - secondly -> if timeframe is more than a month users should use monthly or yearly as those are not consistent. I.e. a month has different time passed depending on which month it is, same goes for year 
* x - monthly
* x - yearly

**Comment by mustermeiszer** [comment]
> Moreover, do I understand correctly that these fees never reduce the reserve because of

yes
